app/models/iterations/project_iterations.rb

```diff
@@ -10,27 +10,49 @@ def current_iteration_start
 
     def past_iterations
       (0...length).map do |iteration_number|
-        start_date = start_date(iteration_number)
-        end_date = end_date(start_date)
-        PastIteration.new(start_date: start_date,
-                          end_date: end_date,
-                          project: @project,
-                          iteration_number: iteration_number + 1)
+        start_at = start_date(iteration_number)
+        end_at = end_date(start_at)
+
+        PastIteration.new(
+          start_date: start_at,
+          end_date: end_at,
+          stories: stories_from(start_at, end_at),
+          iteration_number: iteration_number + 1
+        )
       end
     end
 
     private
 
+    attr_reader :project
+
+    def stories_from(start_at, end_at)
+      stories.select do |story|
+        story.accepted_at >= start_at && story.accepted_at <= end_at
+      end
+    end
+
+    def stories
+      @stories ||= begin
+        project
+          .stories
+          .with_dependencies
+          .where(state: 'accepted')
+          .where.not(accepted_at: nil)
+          .order(:accepted_at)
+      end
+    end
+
     def length
       (days_since_project_start / iteration_length_in_days).floor
     end
 
     def iteration_length_in_days
-      @project.iteration_length * 7
+      project.iteration_length * 7
```

Comment:
Since we're already here, what do you think about extracting the `7` to a `DAYS_IN_A_WEEK` constant?

Reply:
👍